library/std/src/io/stdio.rs

```diff
@@ -434,6 +435,12 @@ impl Stdin {
     }
 }
 
+#[stable(feature = "catch_unwind", since = "1.9.0")]
+impl UnwindSafe for Stdin {}
+
+#[stable(feature = "catch_unwind", since = "1.9.0")]
+impl RefUnwindSafe for Stdin {}
+
```

Comment:
The manual implementation is necessary to avoid a situation like https://github.com/rust-lang/rust/issues/146087.